### ISSUE TITLE
Support Submodules in Terraform Registry

### ISSUE THREAD

**Issue by code-haven** [hashicorp/terraform-cdk#520]
When trying to import a sub-module that's hosted in the terraform registry, the `cdktf get` command fails with `Error: Moved Permanently`. I tried different modules from the registry and the behavior is consistent. Things work fine when importing the root module though.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### cdktf & Language Versions
cdktf: `0.1.0-pre.171`
python: `3.7.6`

### Debug Output

https://gist.github.com/code-haven/da462bb0a3aa6732f05465cb4a21f1ec
### Expected Behavior

Import is successful

### Actual Behavior

Fails with an exception: `Error: Moved Permanently`

### Steps to Reproduce

1. Add a reference to a submodule in `cdktf.json`
```
{
  "terraformModules": ["hashicorp/consul/google//modules/consul-cluster"],
  "codeMakerOutput": "imports",

}

```
2. Run `cdktf get`

### References

* #0000

**Comment by jsteinich** [comment]
Looks like submodules are currently unsupported. They don't even have code generation when referencing the root module.

I don't believe that it would be too difficult to add support.
@code-haven for your use case, would you prefer having to specify each submodule individually, or having all of them generated for the root module?

**Comment by code-haven** [comment]
@jsteinich thanks for the quick response! For my use-case, I don't have a root module. It is a monorepo with 20+ modules organised in a hierarchy with no common interface between them. I suppose I would need the ability to specify the path to individual sub-modules, though not ideal :( 

For reference, this a module that's very similar to the one I'm working on https://registry.terraform.io/modules/claranet/detectors/signalfx/latest

**Comment by skorfmann** [comment]
That's an interesting use-case, thanks for bringing this up. It's somewhat related to https://github.com/hashicorp/terraform-cdk/issues/16

**Comment by skorfmann** [comment]
> For reference, this a module that's very similar to the one I'm working on https://registry.terraform.io/modules/claranet/detectors/signalfx/latest

![Screenshot 2021-01-26 at 15 38 36](https://user-images.githubusercontent.com/136789/105859267-9e701c00-5fec-11eb-99e3-1df7ac22d1df.png)

**Comment by alexjmoore** [comment]
Seeing the same for `"terraform-google-modules/vpn/google//modules/vpn_ha"`, in my view it makes sense to load them all in automatically, or perhaps that could be an option in the `cdktf.json` for a wildcard indicator? e.g. `modulename/*@x.y.z`, if you don't specify the `/*` it just deals with the root.  Just an idea.

**Comment by skorfmann** [comment]
That's resolved with #522
